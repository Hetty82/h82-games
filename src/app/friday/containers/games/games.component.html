<section class="container">
  <h1>Welcome to Friday, the game!</h1>

  <p>
    Here's the current user:
  </p>

  <pre>{{ user$ | async | json }}</pre>

  <p>Choose your game from the list or start a new one:</p>

  <p>
    Here's the games state:
  </p>

  <pre>{{ gamesState$ | async | json }}</pre>

  <button class="button">Create a new game</button>
</section>
