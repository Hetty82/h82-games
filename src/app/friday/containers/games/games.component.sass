@import '~styles/variables'

header
  margin-bottom: $spacing-sm
