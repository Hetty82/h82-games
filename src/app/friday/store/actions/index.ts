export * from './active-game.actions'
export * from './games.actions'
