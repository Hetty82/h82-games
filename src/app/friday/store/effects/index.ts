import { GamesEffects } from './games.effects'
import { InnerGameEffects } from './inner-game.effects'
import { OuterGameEffects } from './outer-game.effects'

export const fridayEffects: any[] = [
  GamesEffects,
  InnerGameEffects,
  OuterGameEffects,
]

export * from './games.effects'
export * from './inner-game.effects'
export * from './outer-game.effects'
