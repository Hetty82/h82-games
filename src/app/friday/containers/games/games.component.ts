import { Component, OnInit } from '@angular/core'
import { select, Store } from '@ngrx/store'

import * as fromStore from '../../store'


@Component({
  selector: 'app-fr-games',
  templateUrl: './games.component.html',
  styleUrls: ['./games.component.sass']
})
export class GamesComponent implements OnInit {
  gamesState$ = this.store.pipe(select(fromStore.getGamesState))
  user$ = this.store.pipe(select(fromStore.getCurrentUser))

  constructor(private store: Store<fromStore.State>) {
  }

  ngOnInit() {
  }

  createGame(event) {
    console.log('let\'s create a game')
  }

  selectGame(event) {
    console.log('let\'s select a game')
  }
}
