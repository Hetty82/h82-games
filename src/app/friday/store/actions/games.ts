import { Action } from '@ngrx/store'

import { Game } from '../../models/game.model'


export type GameError = any
export type GameId = number
export type UserId = number

// Create game
export const CREATE_GAME = '[Friday - Games] Create Game'
export const CREATE_GAME_FAIL = '[Friday - Games] Create Games Fail'
export const CREATE_GAME_SUCCESS = '[Friday - Games] Create Games Success'
// Load games
export const LOAD_GAMES = '[Friday - Games] Load Games'
export const LOAD_GAMES_FAIL = '[Friday - Games] Load Games Fail'
export const LOAD_GAMES_SUCCESS = '[Friday - Games] Load Games Success'
// Select game
export const SELECT_GAME = '[Friday - Games] Select Game'


// Actions
export class CreateGame implements Action {
  readonly type = CREATE_GAME
  constructor(public payload: Game) {}
}
export class CreateGameFail implements Action {
  readonly type = CREATE_GAME_FAIL
  constructor(public payload: GameError) {}
}
export class CreateGameSuccess implements Action {
  readonly type = CREATE_GAME_SUCCESS
  constructor(public payload: Game) {}
}

export class LoadGames implements Action {
  readonly type = LOAD_GAMES
  constructor(public payload: UserId) {}
}
export class LoadGamesFail implements Action {
  readonly type = LOAD_GAMES_FAIL
  constructor(public payload: GameError) {}
}
export class LoadGamesSuccess implements Action {
  readonly type = LOAD_GAMES_SUCCESS
  constructor(public payload: Game[]) {}
}

export class SelectGame implements Action {
  readonly type = SELECT_GAME
  constructor(public payload: GameId) {}
}


// Action types
export type GamesAction =
  | CreateGame
  | CreateGameFail
  | CreateGameSuccess
  | LoadGames
  | LoadGamesFail
  | LoadGamesSuccess
  | SelectGame
