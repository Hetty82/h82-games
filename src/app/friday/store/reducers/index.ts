
import { ActionReducerMap, createFeatureSelector } from '@ngrx/store'

import * as fromActiveGame from './active-game'
import * as fromGames from './games'


export interface State {
  activeGame: fromActiveGame.State
  games: fromGames.State
}

export const reducers: ActionReducerMap<State> = {
  activeGame: fromActiveGame.reducer,
  games: fromGames.reducer,
}

export const getFridayState = createFeatureSelector<State>(
  'friday'
)
