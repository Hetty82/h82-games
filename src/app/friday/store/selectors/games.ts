import { createSelector } from '@ngrx/store'

import * as fromFeature from '../reducers'
import * as fromGames from '../reducers/games'


export const getGamesState = createSelector(fromFeature.getFridayState, (state: fromFeature.State) => state.games)

export const getGameEntities = createSelector(getGamesState, fromGames.getEntities)
export const getGameIds = createSelector(getGamesState, fromGames.getIds)
export const getGamesLoaded = createSelector(getGamesState, fromGames.getLoaded)
export const getGamesLoading = createSelector(getGamesState, fromGames.getLoading)

export const getGames = createSelector(getGameEntities, getGameIds, (entities, ids) => {
  return ids.map(id => entities[id])
})
