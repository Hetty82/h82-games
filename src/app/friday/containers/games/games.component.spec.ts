import { NO_ERRORS_SCHEMA } from '@angular/core'
import { async, ComponentFixture, TestBed } from '@angular/core/testing'
import { Store } from '@ngrx/store'

import { GamesComponent } from './games.component'


describe('GamesComponent', () => {
  let component: GamesComponent
  let fixture: ComponentFixture<GamesComponent>

  beforeEach(async(() => {
    TestBed.configureCompiler({ preserveWhitespaces: false } as any).configureTestingModule({
      declarations: [ GamesComponent ],
      providers: [
        {
          provide: Store,
          useValue: {
            pipe: jest.fn(),
          },
        },
      ],
      schemas: [ NO_ERRORS_SCHEMA ],
    })
    .compileComponents()
  }))

  beforeEach(() => {
    fixture = TestBed.createComponent(GamesComponent)
    component = fixture.componentInstance
    fixture.detectChanges()
  })

  it('should create component', () => {
    expect(component).toBeTruthy()
  })

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot()
  })
})
