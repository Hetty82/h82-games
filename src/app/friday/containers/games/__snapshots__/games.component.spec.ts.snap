// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesComponent should match snapshot 1`] = `
<app-fr-games
  gamesState$="undefined"
  store={[Function Object]}
  user$="undefined"
>
  <section
    class="container"
  >
    <h1>
      Welcome to Friday, the game!
    </h1>
    <p>
       Here's the current user: 
    </p>
    <pre>
      null
    </pre>
    <p>
      Choose your game from the list or start a new one:
    </p>
    <p>
       Here's the games state: 
    </p>
    <pre>
      null
    </pre>
    <button
      class="button"
    >
      Create a new game
    </button>
  </section>
</app-fr-games>
`;
